src/test/munit/tcp-protocol-test-suite.xml

```diff
@@ -0,0 +1,219 @@
+        <logger message="#[payload]" level="INFO"/>
+        <munit-tools:store key='message'/>
+    </flow>
+
+    <flow name="tcp-length-listen">
+        <sockets:listener config-ref="tcp-length-listener"/>
+        <logger message="#[payload]" level="INFO"/>
+        <munit-tools:store key='message'/>
+    </flow>
+
+    <munit:test name="tcp-streaming-test">
+        <munit:enable-flow-sources>
+            <munit:enable-flow-source value="tcp-streaming-listen"/>
+        </munit:enable-flow-sources>
+        <munit:execution>
+            <sockets:send config-ref="tcp-streaming-requester">
+                <sockets:content><![CDATA['Hello world!']]></sockets:content>
+            </sockets:send>
+        </munit:execution>
+        <munit:validation>
+            <until-successful maxRetries="60" millisBetweenRetries="2000">
+                <munit-tools:retrieve key='message'/>
+                <logger message="#[payload]" level="INFO"/>
+                <munit-tools:assert-that
+                        expression="#[payload.^raw as String]"
+                        is="#[MunitTools::containsString('Hello world!')]"
+                        message="Message not received"/>
+            </until-successful>
+        </munit:validation>
+    </munit:test>
+
+    <munit:test name="tcp-reconnect-test">
+        <munit:enable-flow-sources>
+            <munit:enable-flow-source value="tcp-reconnect-listen"/>
+        </munit:enable-flow-sources>
+        <munit:execution>
+            <set-payload value='Hello world!'/>
+            <sockets:send config-ref="tcp-reconnect-requester">
+                <reconnect count="10"/>
+            </sockets:send>
+            <sockets:send config-ref="tcp-reconnect-requester">
+                <reconnect count="10"/>
+            </sockets:send>
+            <set-payload value=''/>
+        </munit:execution>
+        <munit:validation>
+            <until-successful maxRetries="60" millisBetweenRetries="2000">
+                <munit-tools:retrieve key='message'/>
+                <logger message="#[payload]" level="INFO"/>
+                <munit-tools:assert-that
+                        expression="#[payload.^raw as String]"
+                        is="#[MunitTools::containsString('Hello world!')]"
+                        message="Message not received"/>
+            </until-successful>
+        </munit:validation>
+    </munit:test>
+
+    <munit:test name="tcp-length-longer-than-15-test" expectedErrorType="SOCKETS:LENGTH_EXCEEDED">
+        <munit:enable-flow-sources>
+            <munit:enable-flow-source value="tcp-length-listen"/>
+        </munit:enable-flow-sources>
+        <munit:execution>
+            <sockets:send config-ref="tcp-length-requester">
+                <sockets:content><![CDATA['Hello world!Hello world!Hello world!Hello world!']]></sockets:content>
+            </sockets:send>
+        </munit:execution>
+        <munit:validation>
```

Comment:
I don't think the validation section is needed if we are expecting an error

Reply:
Error is not thrown immediately so loop is good there. Otherwise test can stop without error because listener is not started.